### PR TITLE
pro-lxd

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Why is this needed?

This implements 3 features all towards the goal of auto-attaching lxd guests as described in spec US110.

1. `pro api u.pro.attach.guest.get_guest_token.v1`
2. `pro config set lxd_guest_attach`
3. `pro auto-attach` within a lxd container

Builds of this PR for jammy and noble are in this PPA https://launchpad.net/~orndorffgrant/+archive/ubuntu/pro-client-lxd-testing

Docs will be in #3334

## Test Steps

Try out the new tests in `lxd.feature`. 

---

- [ ] *(un)check this to re-run the checklist action*

### CODE REVIEW

File: uaclient/cli/config.py [hunk 1]
```diff
@@ -180,6 +182,14 @@ def action_config_set(args, *, cfg, **kwargs):
...
         else:
             state_files.apt_news_contents_file.delete()
+    elif set_key == "lxd_guest_attach":
+        if not _is_attached(cfg).is_attached:
+            raise exceptions.UnattachedError()
+        state_files.lxd_pro_config_file.write(
+            state_files.LXDProConfig(
+                guest_attach=LXDGuestAttachEnum.from_value(set_value.lower())
+            )
+        )
```
Comment (orndorffgrant) [inline uaclient/cli/config.py:192]: Needs equivalent in process_config

**Comment by lucasmoura** [comment]
@orndorffgrant The PR LGTM, but there are some valid integration test failures here, such as `u.pro.config.v1` and `Help text for the Pro Client commands`

**Comment by orndorffgrant** [comment]
As part of our process to get ready for v35, I have rebased `next-v35` onto `main`. As a result, `next-v35` is now obsolete. Please rebase this on `main` and target the PR to `main`.

**Comment by orndorffgrant** [comment]
@lucasmoura The lxd.feature tests should be fixed now - thanks for noticing that problem

**Review by renanrodrigo — APPROVED** [review]

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**